File: python/Neetcode/Arrays/Medium/179.LargestNumber.py
# Given a list of non-negative integers nums, 
# arrange them such that they form the largest number and return it.
# Since the result may be very large, so you need to return a string instead of an integer.


def largestNumber(nums):
    for i, number in enumerate(nums):
        nums[i] = str(number)

    def compare(n1, n2):
        if n1+ n2 > n2 + n1:
            return -1
        else:
            return 1
        
    nums = sorted(nums, key=cmp_to_key(compare))

    return str(int("".join(nums)))

print(largestNumber([21,9]))

File: swift/Neetcode/Arrays/Medium/179.LargestNumber.swift
// Given a list of non-negative integers nums, 
// arrange them such that they form the largest number and return it.
// Since the result may be very large, so you need to return a string instead of an integer.

//lexicographically
func largestNumber(_ nums: [Int]) -> String {
    var nums = nums.map { String($0) }.sort { return $0 + $1 > $1 + $0 }
    
    if nums[0] == "0" { return "0" }
    
    return nums.joined()
}

print(largestNumber([21,9]))